packages/next/src/build/normalize-catchall-routes.ts

```diff
@@ -12,15 +12,15 @@ export function normalizeCatchAllRoutes(
   normalizer = new AppPathnameNormalizer()
 ) {
   const catchAllRoutes = [
-    ...new Set(Object.values(appPaths).flat().filter(isCatchAllRoute)),
+    ...new Set(Object.values(appPaths).flat().filter(isCatchAllRoute).sort((a,b)=> b.split("/").length - a.split("/").length)),
```

Comment:
mind adding a comment here that sorting is important because we want to match the most specific path?

Reply:
@feedthejim I have added a comment.
https://github.com/vercel/next.js/pull/58368/commits/4922a630f40635b8f43e78e45fe4ab198b159fb1